components/nimbus/tests/test_restart.rs

```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+mod common;
+#[cfg(feature = "rkv-safe-mode")]
+#[cfg(test)]
+mod test {
+
+    use super::common::new_test_client_with_db;
+    #[cfg(feature = "rkv-safe-mode")]
+    use nimbus::error::Result;
+    use tempdir::TempDir;
+
+    #[test]
+    fn test_restart_opt_in() -> Result<()> {
+        let temp_dir = TempDir::new("test_restart_opt_in")?;
+        let client = new_test_client_with_db(&temp_dir)?;
+        client.initialize()?;
+        client.fetch_experiments()?;
```

Comment:
I think I'd use `set_experiments_locally(payload: string)` to get couple of experiments which you can cause trigger the issue.

I think the `secure-gold` experiment in the `test/experiments` folder is set to always enroll: you need an experiment that won't enroll naturally.

You _might_ be able add another file to the `tests/experiments` directory, but this might be more trouble than it's worth.

Reply:
I think you'll need one experiment that has `"targeting": "false"` and one with a `bucketConfig` as:

```json
  "bucketConfig": {
    "count": 0,
    "start": 0,
    "total": 10000,
    "namespace": "secure-gold",
    "randomizationUnit": "nimbus_id"
  },
```

i.e. 0 out of 10000 targeted devices will be enrolled into the experiment.

The `targeting` *string* is a JEXL string which evaluates to a boolean. If it evaluates to false then the client is not enrolled into the experiment.